be/src/olap/reader.cpp

```diff
@@ -467,7 +467,7 @@ OLAPStatus Reader::_capture_rs_readers(const ReaderParams& read_params) {
         rs_readers = &read_params.rs_readers;
     } else {
         _tablet->obtain_header_rdlock();
-        _tablet->capture_rs_readers(_version, &_own_rs_readers);
+        RETURN_NOT_OK(_tablet->capture_rs_readers(_version, &_own_rs_readers));
         _tablet->release_header_lock();
 
```

Comment:
if not return ok, it will return if own_res_readers's size < 1. Is that ok?

Reply:
Here return not ok means that the reader init failed and the reading will return failed directly. I think it is a right process.